src/vulkan/vulkan_initializers.cpp

```diff
@@ -163,4 +163,26 @@ namespace IC {
 
         return pipeline;
     }
+
+    // ImGui
+    void InitImGui(VulkanDevice &device, GLFWwindow *window, VkDescriptorPool descriptorPool, VkFormat imageFormat) {
+        ImGui::CreateContext();
+        ImGui_ImplGlfw_InitForVulkan(window, true);
+
+        ImGui_ImplVulkan_InitInfo initInfo = {};
+        initInfo.Instance = device.Instance();
+        initInfo.PhysicalDevice = device.PhysicalDevice();
+        initInfo.Device = device.Device();
+        initInfo.Queue = device.GraphicsQueue();
+        initInfo.DescriptorPool = descriptorPool;
+        initInfo.MinImageCount = 3;
+        initInfo.ImageCount = 3;
+        initInfo.UseDynamicRendering = true;
+        initInfo.MSAASamples = VK_SAMPLE_COUNT_1_BIT;
+        initInfo.ColorAttachmentFormat = imageFormat;
+
+        ImGui_ImplVulkan_Init(&initInfo, VK_NULL_HANDLE);
+        ImGui_ImplVulkan_CreateFontsTexture();
+        ImGui_ImplVulkan_DestroyFontsTexture();
```

Comment:
What are these two functions doing? Creating the Destroying the Fonts texture?

Reply:
Apparently neither of these are needed as it's done behind the scenes now 
```
2023-11-10: *BREAKING CHANGE*: Removed parameter from ImGui_ImplVulkan_CreateFontsTexture(): backend now creates its own command-buffer to upload fonts.
//              *BREAKING CHANGE*: Removed ImGui_ImplVulkan_DestroyFontUploadObjects() which is now unecessary as we create and destroy those objects in the backend.
//              ImGui_ImplVulkan_CreateFontsTexture() is automatically called by NewFrame() the first time.
//              You can call ImGui_ImplVulkan_CreateFontsTexture() again to recreate the font atlas texture.
//              Added ImGui_ImplVulkan_DestroyFontsTexture() but you probably never need to call this.
```